builtin/logical/ad/backend.go

```diff
@@ -1 +1,25 @@
 package ad
+
+import (
+	"context"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func Factory(_ context.Context, conf *logical.BackendConfig) (logical.Backend, error) {
+
+	confHandler := &configHandler{conf.Logger}
```

Comment:
Generally, other endpoints will need to access the config object. Do you have a plan for caching the config and making sure other endpoints can access it?

Reply:
I was _thinking_ about caching it but I didn't on this iteration. However, now that you point it out, I do think it would make sense. I could just keep it in-memory on the `configHandler` and invalidate it when this and that call come in. I'll do that.

Reply:
I was originally planning to pass the `configHandler` object to each of them so they'd have access to the config. However, adding a config cache has led me to a cleaner design where I'll just pass a read-only version of the config cache to the other handlers. I haven't pushed it up yet, but will soon.